pyanaconda/modules/storage/storage.py

```diff
@@ -71,87 +72,60 @@ def __init__(self):
         self.partitioning_reset = Signal()
 
         # Initialize modules.
-        self._modules = []
-
-        self._storage_checker_module = StorageCheckerModule()
-        self._add_module(self._storage_checker_module)
-
-        self._device_tree_module = DeviceTreeModule()
-        self._add_module(self._device_tree_module)
-
-        self._disk_init_module = DiskInitializationModule()
-        self._add_module(self._disk_init_module)
-
-        self._disk_selection_module = DiskSelectionModule()
-        self._add_module(self._disk_selection_module)
-
-        self._snapshot_module = SnapshotModule()
-        self._add_module(self._snapshot_module)
-
-        self._bootloader_module = BootloaderModule()
-        self._add_module(self._bootloader_module)
-
-        self._fcoe_module = FCOEModule()
-        self._add_module(self._fcoe_module)
-
-        self._iscsi_module = ISCSIModule()
-        self._add_module(self._iscsi_module)
-
-        self._nvdimm_module = NVDIMMModule()
-        self._add_module(self._nvdimm_module)
-
-        self._dasd_module = DASDModule()
-        self._add_module(self._dasd_module)
-
-        self._zfcp_module = ZFCPModule()
-        self._add_module(self._zfcp_module)
+        self._modules = SubmoduleManager()
+        self._modules.add_module("storage_checker", StorageCheckerModule())
+        self._modules.add_module("device_tree", DeviceTreeModule())
+        self._modules.add_module("disk_init", DiskInitializationModule())
+        self._modules.add_module("disk_selection", DiskSelectionModule())
+        self._modules.add_module("snapshot", SnapshotModule())
+        self._modules.add_module("bootloader", BootloaderModule())
+        self._modules.add_module("fcoe", FCOEModule())
+        self._modules.add_module("iscsi", ISCSIModule())
+        self._modules.add_module("nvdimm", NVDIMMModule())
+        self._modules.add_module("dasd", DASDModule())
+        self._modules.add_module("zfcp", ZFCPModule())
 
         # Connect modules to signals.
         self.storage_changed.connect(
-            self._device_tree_module.on_storage_changed
+            self._modules["device_tree"].on_storage_changed
```

Comment:
Btw. I planned to introduce a new class that would define the `on_storage_changed` method to replace most of these statements with a single loop.

Reply:
Ah, yes, that makes sense. How about:

1. Define another mixin `StorageSubscripter` with abstract method `on_storage_changed`
2. Add this as another parent to all the submodules of interest.
3. Loop over all submodules, filter by the mixin class, connect the signals.

Reply:
Actually, no, that would not be a mixin but a regular ancestor class. I got confused after looking at `DeviceTreeModule` with its rich ancestry.

Reply:
Never mind, pushed here, too.

Reply:
I think the mixin would make a lot of sense. Btw. I don't know why `DeviceTreeModule` isn't just `BaseModule`.